src/fu-engine-helper.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2020 Mario Limonciello <mario.limonciello@dell.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#define G_LOG_DOMAIN				"FuEngine"
+
+#include "config.h"
+
+#include "fu-engine.h"
+#include "fu-engine-helper.h"
+
+gboolean
+fu_engine_update_motd (FuEngine *self, GError **error)
+{
+	guint upgrade_count = 0;
+	g_autoptr(GPtrArray) devices = NULL;
+	g_autofree gchar *target = NULL;
+        g_autoptr(GString) str = g_string_new ("");
+
+	/* get devices from daemon */
+	devices = fu_engine_get_devices (self, NULL);
+	if (devices == NULL)
+		return TRUE;
+
+	for (guint i = 0; i < devices->len; i++) {
+		FwupdDevice *dev = g_ptr_array_index (devices, i);
+		g_autoptr(GPtrArray) rels = NULL;
+
+		/* get the releases for this device and filter for validity */
+		rels = fu_engine_get_upgrades (self,
+					       fwupd_device_get_id (dev),
+					       NULL);
+		if (rels == NULL)
+			continue;
+		upgrade_count++;
+	}
+
+	/* If running under systemd unit, use the directory as a base */
+	if (g_getenv ("RUNTIME_DIRECTORY") != NULL) {
+		target = g_build_filename (g_getenv ("RUNTIME_DIRECTORY"),
+					   PACKAGE_NAME,
+					   MOTD_FILE,
+					   NULL);
+	/* otherwise use the cache directory */
+	} else {
+		g_autofree gchar *directory = fu_common_get_path (FU_PATH_KIND_CACHEDIR_PKG);
+		target = g_build_filename (directory, MOTD_DIR, MOTD_FILE, NULL);
+	}
+
+	/* create the directory and file, even if zero devices; we want an empty file then */
+	if (!fu_common_mkdir_parent (target, error))
+		return FALSE;
+	switch (upgrade_count) {
+	case 0:
+		break;
+	case 1:
+                g_string_append_printf (str, "%u device has a firmware upgrade available. "
+					     "Run `fwupdmgr get-upgrades` for more information\n",
+					     upgrade_count);
+		break;
+	default:
+		g_string_append_printf (str, "%u devices have a firmware upgrade available. "
+					     "Run `fwupdmgr get-upgrades` for more information\n",
```

Comment:
could this `Run fwupdmgr get-upgrades for more information` be split out below? I think you also prbably want to mark it for translation, and to add fu-engine-helper.c to POTFILES.